core/gotemplate/templatefunctions_test.go

```diff
@@ -0,0 +1,220 @@
+package gotemplate
+
+import (
+	"context"
+	"fmt"
+	"html/template"
+	"net/url"
+	"reflect"
+	"sort"
+	"testing"
+)
+
+type MockRouter struct{}
+
+func (m MockRouter) Relative(name string, params map[string]string) (*url.URL, error) {
+	return url.Parse(fmt.Sprintf("http://name-%v.com/param-amount-%d/", name, len(params)))
+}
+
+func (m MockRouter) Data(ctx context.Context, handler string, params map[interface{}]interface{}) interface{} {
+	var stringParams []string
+	for key, value := range params {
+		stringParams = append(stringParams, fmt.Sprintf("%v:%v", key, value))
+	}
+
+	sort.Strings(stringParams)
+
+	return fmt.Sprintf("%v %v", handler, stringParams)
+}
+
+var _ urlRouter = MockRouter{}
+
+func Test_dataFunc_Func(t *testing.T) {
+	type args struct {
+		what   string
+		params []string
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want interface{}
+	}{
+		{
+			name: "no params",
+			args: args{
+				what: "test",
+			},
+			want: "test []",
+		},
+		{
+			name: "valid params set",
+			args: args{
+				what:   "test",
+				params: []string{"key-1", "value-1", "key-2", "value-2"},
+			},
+			want: "test [key-1:value-1 key-2:value-2]",
+		},
+		{
+			name: "invalid params set",
+			args: args{
+				what:   "test",
+				params: []string{"key-1"},
+			},
+			want: "test []",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			df := &dataFunc{}
+			df.Inject(MockRouter{})
+
+			templateFunction := df.Func(context.Background()).(func(what string, params ...string) interface{})
+			if got := templateFunction(tt.args.what, tt.args.params...); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("dataFunc() = %q, want %q", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_getFunc_Func(t *testing.T) {
```

Comment:
This test is the same as `Test_dataFunc_Func`, with the only difference in calling `get` instead of `data`, which is just an alias for `data`.
Remove this test and call both `get` and `data` in the `Test_dataFunc_Func` function, otherwise these tests won't be maintainable (duplicate code).